### ISSUE TITLE
Hook BackoffLimitExceeded even though job completes successfully

### ISSUE THREAD

**Issue by vmalloc** [helm/helm#6873]
I have a pre-upgrade hook that looks like this in my chart:
```
apiVersion: batch/v1
kind: Job
metadata:
  name: "{{.Release.Name}}-migrate"
  labels:
    app.kubernetes.io/managed-by: {{.Release.Service | quote }}
    app.kubernetes.io/instance: {{.Release.Name | quote }}
    app.kubernetes.io/version: {{ .Chart.AppVersion | quote}}
    helm.sh/chart: "{{.Chart.Name}}-{{.Chart.Version}}"
  annotations:
    "helm.sh/hook": "pre-upgrade,pre-install"
    "helm.sh/hook-weight": "-1"
    "helm.sh/hook-delete-policy": "before-hook-creation"
spec:
  ttlSecondsAfterFinished: 300
  activeDeadlineSeconds: 300
  template:
    metadata:
      name: "{{.Release.Name}}"
      labels:
        app.kubernetes.io/managed-by: {{.Release.Service | quote }}
        app.kubernetes.io/instance: {{.Release.Name | quote }}
        helm.sh/chart: "{{.Chart.Name}}-{{.Chart.Version}}"
    spec:
      restartPolicy: Never
      containers:
      - name: migrate-db
        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
        imagePullPolicy: Always
        command: ["..."]
        env:
          ...
```
I am running the upgrade and getting 
```
debug] Created tunnel using local port: '54226'

[debug] SERVER: "127.0.0.1:54226"

UPGRADE FAILED
Error: Job failed: BackoffLimitExceeded
Error: UPGRADE FAILED: Job failed: BackoffLimitExceeded
```

In the tiller log I also see indication that helm thinks the job failed:
```
warning: Release home-backend pre-upgrade home-backend/templates/migrate.yaml could not complete: Job failed: BackoffLimitExceeded
```

However the job succeeded and appears just fine in `pod describe`. I am also pretty sure error gets returned even before the job completes successfully...

Output of `helm version`:
```
Client: &version.Version{SemVer:"v2.15.2", GitCommit:"8dce272473e5f2a7bf58ce79bb5c3691db54c96b", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.15.2", GitCommit:"8dce272473e5f2a7bf58ce79bb5c3691db54c96b", GitTreeState:"clean"}
```

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T23:42:50Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.10-eks-5ac0f1", GitCommit:"5ac0f1d9ab2c254ea2b0ce3534fd72932094c6e1", GitTreeState:"clean", BuildDate:"2019-08-20T22:39:46Z", GoVersion:"go1.11.13", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.):  EKS

**Comment by vmalloc** [comment]
I found out that I had other (unrelated) jobs that were pretty old and did not run to completion. Those were other cron jobs that *are not* helm hooks. Once I deleted those jobs this error disappeared.

I still believe this is a bug though - failures in unrelated jobs should not cause an upgrade to fail...

**Comment by lisfox1** [comment]
Removing failed jobs fixed the issue for me as well. Thank you @vmalloc

**Comment by froch** [comment]
same for us, deleting failed jobs from another unrelated deployment "fixed" this. thank you

**Comment by walbertus** [comment]
This also to happen to us, thanks @vmalloc

**Comment by vmalloc** [comment]
It seems like this, along with  #6873, are two very painful regressions with 2.15.x... I wonder if we could get any of the developers to comment on plans to address these issues...

**Comment by bacongobbler** [comment]
If you can identify the issue we'd be happy to review any patches for 2.15. Right now we're focusing on getting Helm 3 released so we haven't had time to look at this.

**Comment by jorge-gasca** [comment]
This is possibly fixed by #6907
